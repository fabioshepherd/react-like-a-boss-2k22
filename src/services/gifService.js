import axios from 'axios';

const key = 'OTWW1XT3O4FI';

export const getGifs = async (q, limit) => {
  const gifResponse = await axios.get('https://g.tenor.com/v1/search', {
    params: {
      q,
      limit,
      key,
    },
  });

  return gifResponse;
};

export const getRandomGifs = async (q, limit) => {
  const gifResponse = await axios.get('https://g.tenor.com/v1/random', {
    params: {
      q,
      limit,
      key,
    },
  });

  return gifResponse;
};
