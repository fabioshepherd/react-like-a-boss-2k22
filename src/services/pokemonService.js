import axios from 'axios';

export const getPokemonList = async (limit) => {
  const pokemonResponse = await axios.get(
    'https://pokeapi.co/api/v2/pokemon/',
    {
      params: {
        limit,
      },
    }
  );

  return pokemonResponse;
};

export const getPokemonByNumber = async (number) => {
  const pokemonResponse = await axios.get(
    `https://pokeapi.co/api/v2/pokemon/${number}`
  );

  return pokemonResponse;
};
