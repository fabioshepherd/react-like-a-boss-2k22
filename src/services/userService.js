import axios from 'axios';

export const getRandomUsers = async (results) => {
  const usersResponse = await axios.get('https://randomuser.me/api/', {
    params: {
      results,
    },
  });

  return usersResponse;
};
