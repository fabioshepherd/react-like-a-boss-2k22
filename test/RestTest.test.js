import { getPokemonList } from "../src/services/pokemonService";

it("Rest Test", () => {
    getPokemonList().then((result) => {
        console.log(result)
    })
});