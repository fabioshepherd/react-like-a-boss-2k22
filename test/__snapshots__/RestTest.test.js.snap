// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rest Test 1`] = `
Object {
  "config": Object {
    "adapter": [Function],
    "data": undefined,
    "env": Object {
      "FormData": [Function],
    },
    "headers": Object {
      "Accept": "application/json, text/plain, */*",
    },
    "maxBodyLength": -1,
    "maxContentLength": -1,
    "method": "get",
    "params": Object {
      "limit": undefined,
    },
    "timeout": 0,
    "transformRequest": Array [
      [Function],
    ],
    "transformResponse": Array [
      [Function],
    ],
    "transitional": Object {
      "clarifyTimeoutError": false,
      "forcedJSONParsing": true,
      "silentJSONParsing": true,
    },
    "url": "https://pokeapi.co/api/v2/pokemon/",
    "validateStatus": [Function],
    "xsrfCookieName": "XSRF-TOKEN",
    "xsrfHeaderName": "X-XSRF-TOKEN",
  },
  "data": Object {
    "count": 1126,
    "next": "https://pokeapi.co/api/v2/pokemon/?offset=20&limit=20",
    "previous": null,
    "results": Array [
      Object {
        "name": "bulbasaur",
        "url": "https://pokeapi.co/api/v2/pokemon/1/",
      },
      Object {
        "name": "ivysaur",
        "url": "https://pokeapi.co/api/v2/pokemon/2/",
      },
      Object {
        "name": "venusaur",
        "url": "https://pokeapi.co/api/v2/pokemon/3/",
      },
      Object {
        "name": "charmander",
        "url": "https://pokeapi.co/api/v2/pokemon/4/",
      },
      Object {
        "name": "charmeleon",
        "url": "https://pokeapi.co/api/v2/pokemon/5/",
      },
      Object {
        "name": "charizard",
        "url": "https://pokeapi.co/api/v2/pokemon/6/",
      },
      Object {
        "name": "squirtle",
        "url": "https://pokeapi.co/api/v2/pokemon/7/",
      },
      Object {
        "name": "wartortle",
        "url": "https://pokeapi.co/api/v2/pokemon/8/",
      },
      Object {
        "name": "blastoise",
        "url": "https://pokeapi.co/api/v2/pokemon/9/",
      },
      Object {
        "name": "caterpie",
        "url": "https://pokeapi.co/api/v2/pokemon/10/",
      },
      Object {
        "name": "metapod",
        "url": "https://pokeapi.co/api/v2/pokemon/11/",
      },
      Object {
        "name": "butterfree",
        "url": "https://pokeapi.co/api/v2/pokemon/12/",
      },
      Object {
        "name": "weedle",
        "url": "https://pokeapi.co/api/v2/pokemon/13/",
      },
      Object {
        "name": "kakuna",
        "url": "https://pokeapi.co/api/v2/pokemon/14/",
      },
      Object {
        "name": "beedrill",
        "url": "https://pokeapi.co/api/v2/pokemon/15/",
      },
      Object {
        "name": "pidgey",
        "url": "https://pokeapi.co/api/v2/pokemon/16/",
      },
      Object {
        "name": "pidgeotto",
        "url": "https://pokeapi.co/api/v2/pokemon/17/",
      },
      Object {
        "name": "pidgeot",
        "url": "https://pokeapi.co/api/v2/pokemon/18/",
      },
      Object {
        "name": "rattata",
        "url": "https://pokeapi.co/api/v2/pokemon/19/",
      },
      Object {
        "name": "raticate",
        "url": "https://pokeapi.co/api/v2/pokemon/20/",
      },
    ],
  },
  "headers": Object {
    "cache-control": "public, max-age=86400, s-maxage=86400",
    "content-length": "307",
    "content-type": "application/json; charset=utf-8",
  },
  "request": XMLHttpRequest {},
  "status": 200,
  "statusText": "OK",
}
`;
